scripts/dts/extract/clocks.py

```diff
@@ -42,27 +81,34 @@ def _extract_consumer(self, node_address, yaml, clocks, names, def_label):
                                     str(clock_provider)))
                 clock_provider_node_address = phandles[cell]
                 clock_provider = reduced[clock_provider_node_address]
-                clock_provider_compat = get_compat(clock_provider_node_address)
-                clock_provider_bindings = yaml[clock_provider_compat]
-                clock_provider_label = self.get_node_label_string( \
-                                                clock_provider_node_address)
-                nr_clock_cells = int(clock_provider['props'].get(
-                                     '#clock-cells', 0))
-                clock_cells_string = clock_provider_bindings.get(
-                    'cell_string', 'CLOCK')
-                clock_cells_names = clock_provider_bindings.get(
-                    '#cells', ['ID', 'CELL1',  "CELL2", "CELL3"])
+                clock_nr_cells = int(clock_provider['props'].get('#clock-cells', 0))
                 clock_cells = []
             else:
                 clock_cells.append(cell)
             clock_cell_index += 1
-            if clock_cell_index > nr_clock_cells:
-                # clock consumer device - clocks info
-                #####################################
+            if clock_cell_index > clock_nr_cells:
+
+                 # generate EDTS
+                self._edts_insert_clock_cells(node_address, yaml,
```

Comment:
I think we should somehow also insert the clock controller reference in the node clocks.
For instance:
```
            "clocks": {
                "0": {
                    "controller": "/soc/rcc@40021000",
                    "bus": 1,
                    "bits": 2
                }
            },
```
This could/should be done the same way for GPIOs and every property for which a property-controller is defined.

Reply:
> ```"controller": "/soc/rcc@40021000",```

This is the reference to the clock provider.  It should not be mixed with the <device>-controller directive (e.g. the clock-controller directive is within the rcc node). So better name it clock-provider, like in the clock bindings.

Reply:
I could not find reference to "provider" in bindings, can you elaborate?

"/soc/rcc@40021000" having "clock-controller" property, it is then a clock controller. When referring to it, for this capacity of clock control, it makes sense to me to use "controller". Introducing "provider" for clocks and "xx" for gpio will create confusion IMHO. Having one keyword will ease integration and adoption.